### ISSUE TITLE
compiler crashes with floating point exception

### ISSUE THREAD

**Issue by dneto0** [google/clspv#37]
Occurs when compiling this code:
```
typedef struct {
  int a, b;
} S;

S convert(int n) {
  S s = { n, n};
  return s;
}

kernel void foo(global S* A, global uchar4* B, int n) {
 *B = (uchar4)((uchar)n,1,2,3);
 *A = convert(10);
}

```
Does not occur when S gains two extra int members, and the convert function is adjusted accordingly.

**Comment by dneto0** [comment]
Another test case:
```
typedef struct {
  uint x;
  uint y;
} S;

kernel void foo(global S *A, int n) {
  S before = *A;
  before.y += 2;
  *A = before;
}
```
It's a divide by zero exception:

Program received signal SIGFPE, Arithmetic exception.
0x000000000050d072 in (anonymous namespace)::ReplacePointerBitcastPass::runOnModule (this=0x5fd0bb0, M=...) at /lib/ReplacePointerBitcastPass.cpp:761
761                 unsigned NumElement = DstTyBitWidth / SrcTyBitWidth;

DstTyBitWidth is 64, SrcTyBitWidth is 0

**Comment by dneto0** [comment]
Running Instcombine results in 

  store i64 %4, i64 addrspace(1)* %3, align 4

So the source pointee type is a 64bit scalar.  But it looks like that part of the code generator is expecting a vector.

**Comment by alan-baker** [comment]
Working as of 00e7a58a2834ee938830f82b42a0692fc2232511 at least. I believe it was fixed by #370.
